cirq-core/cirq/circuits/moment.py

```diff
@@ -192,6 +192,9 @@ def with_operations(self, *contents: 'cirq.OP_TREE') -> 'cirq.Moment':
         """
         flattened_contents = tuple(op_tree.flatten_to_ops(contents))
 
+        if not flattened_contents:
+            return self
```

Comment:
Please add a unit test of this behavior of `Moment.with_operations`

Reply:
Done.